File: find-pivot-index/find-pivot-index.java
class Solution {
    public int pivotIndex(int[] nums) {
        int sum=0;
        int left=0;
        for(int x: nums){sum+=x;}
        for(int i=0; i<nums.length;i++){

            if(sum-(left+nums[i])==left){
                return i;
            }
            left+=nums[i];
        }
        return -1;
    }
}

File: 724-find-pivot-index/find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        
        prefix = [0 for num in range(len(nums))]

        postfix = [0 for num in range(len(nums))]

        total = 0

        for index in range(len(nums)):
            prefix[index] = total
            total += nums[index]

        total = 0

        for index in range(len(nums) - 1, -1, -1):
            postfix[index] = total
            total += nums[index]

        for index in range(len(nums)):
            if prefix[index] == postfix[index]:
                return index

        return -1
        